ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -112,22 +149,45 @@ function Set-RsDatabase
             $username = $rsWmiObject.WindowsServiceIdentityActual
             $password = ''
         }
-        
+
         else
         {
-            if ($DatabaseCredential -eq $null)
+            if ($DatabaseCredential -eq $nul)
             {
                 throw "No Database Credential specified! Database credential must be specified when configuring $DatabaseCredentialType authentication."
             }
             $username = $DatabaseCredential.UserName
             $password = $DatabaseCredential.GetNetworkCredential().Password
         }
         #endregion Validating authentication and normalizing credentials
-        
+
+        #region Validating admin authentication and normalizing credentials
+        $adminUsername = ''
+        $adminPassword = $null
+        # default is service account
+        if ($AdminDatabaseCredentialType -like 'serviceaccount' -or -not $AdminDatabaseCredential)
+        {
+            $adminUsername = $rsWmiObject.WindowsServiceIdentityActual
+            $adminPassword = ''
+        }
+
+        else
+        {
+            if ($AdminDatabaseCredential -eq $null)
+            {
+                throw "No Admin Database Credential specified! Admin Database credential must be specified when configuring $DatabaseCredentialType authentication."
+            }
+            $adminUsername = $AdminDatabaseCredential.UserName
+            $adminPassword = $AdminDatabaseCredential.GetNetworkCredential().Password
+        }
+        $isWindowsAdminAccount = ($AdminDatabaseCredentialType -like "Windows") -or ($AdminDatabaseCredentialType -like "ServiceAccount")
```

Comment:
The exception is SQL credentials , I suggest have a variable $isSQLCredentials which gets only to true in the if block where you validate if that is the value passed

Reply:
Heh, make it all go reverse! Good thing I have find-replace. :-)